src/main.c

```diff
@@ -807,14 +859,26 @@ static void handler_MonomeGridKey(s32 data) {
 					keyfirst_pos = x;
 				}
 				else if(key_alt == 1) {
-					if(x == LENGTH)
+					if(x == LENGTH) {
 						w.wp[pattern].step_mode = mForward;
-					else if(x == LENGTH-1)
+                    }
+					else if(x == LENGTH-1) {
 						w.wp[pattern].step_mode = mReverse;
-					else if(x == LENGTH-2)
+                    }
+					else if(x == LENGTH-2) {
 						w.wp[pattern].step_mode = mDrunk;
-					else if(x == LENGTH-3)
+                    }
+					else if(x == LENGTH-3) {
 						w.wp[pattern].step_mode = mRandom;
+                    }
+					else if(x == LENGTH-4) {
+						w.wp[pattern].step_mode = mPing;
+                        w.wp[pattern].ping_dir = mPingFwd;
+                    }
+					else if(x == LENGTH-5) {
```

Comment:
The enums are 0 up, so should be easy enough. If we don't mind the overhead of setting the mPingFwd regardless of mode, we might be able to combine all of the modes into a single block (LENGTH-x).